File: algorithms/0718.maximum_length_of_repeated_subarray/maximum_length_of_repeated_subarray.cpp
class Solution {
public:
    int findLength(vector<int>& nums1, vector<int>& nums2) {
        int result = 0;
        vector<int> dp(nums2.size() + 1, 0);
        for (size_t i = 1; i <= nums1.size(); ++i) {
            for (size_t j = nums2.size(); j > 0; --j) {
                if (nums1[i - 1] == nums2[j - 1]) {
                    dp[j] = dp[j - 1] + 1;
                } else {
                    dp[j] = 0;
                }
                result = max(dp[j], result);
            }
        }
        return result;
    }
};

File: algorithms/0718.maximum_length_of_repeated_subarray/maximum_length_of_repeated_subarray.rs
impl Solution {
    pub fn find_length(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
        let (m, n) = (nums1.len(), nums2.len());
        let mut dp = vec![0; n + 1];
        let mut last = vec![0; n + 1];
        let mut result = 0;
        for i in 1..=m {
            dp.swap_with_slice(&mut last);
            for j in 1..=n {
                if nums1[i - 1] == nums2[j - 1] {
                    dp[j] = last[j - 1] + 1;
                    result = result.max(dp[j]);
                } else {
                    dp[j] = 0;
                };
            }
        }
        result
    }
}